monkey/monkey_island/cc/repository/file_agent_log_repository.py

```diff
@@ -0,0 +1,35 @@
+import io
+
+from monkey_island.cc import repository
+from monkey_island.cc.models import AgentID
+from monkey_island.cc.repository import (
+    IAgentLogRepository,
+    IFileRepository,
+    RetrievalError,
+    UnknownRecordError,
+)
+
+AGENT_LOG_FILE_NAME_PREFIX = "agent_log_"
+
+
+class FileAgentLogRepository(IAgentLogRepository):
+    def __init__(self, file_repository: IFileRepository):
+        self._file_repository = file_repository
+
+    def upsert_agent_log(self, agent_id: AgentID, log_contents: str):
+        self._file_repository.save_file(
+            f"{AGENT_LOG_FILE_NAME_PREFIX}{agent_id}", io.BytesIO(log_contents.encode())
```

Comment:
this filename is not human readable. What we have in the database IS human readable, because the logs are next to the agent information. The front end does `let filename = res['monkey_label'].split(':').join('-') + ' - ' + timestamp + '.log';`. 
The timestamp is useless(os already tracks it).
The prefix is also useless, you can tell it's agent log by the folder name of the repository.
There should be a string to identify the agent in human-readable format. I think the best name would be the label (whatever we see in the map) + iteration(which run of monkey on that machine this log contains).

Reply:
The prefix is valuable, since files other than agent logs may be stored in the `IFileRepository`. The file names should probably end in `.log`.

`agent-{agent_id}.log` should get the job done. If we'd prefer to use something in addition to the agent ID to identify machines, we can look it up in the `IAgentRepository`, but IDK that it's necessary. Regardless, agent ID needs to be included, because IPs, hostnames, or other "identifying" fields might be duplicated across multiple machines.

Reply:
So agent logs won't have their own dedicated dir?

Reply:
This implementation doesn't put logs in their own dedicated directory. If we feel that's necessary down the road, we can look at making changes.